ang/civicase/case/details/people-tab/directives/case-roles-tab.directive.js

```diff
@@ -601,6 +609,52 @@
       }
     }
 
+    /**
+     * @param {ContactPromptResult} contactPromptResult the contact returned by the confirm dialog
+     * @returns {boolean} if contact is validated
+     */
+    function validateContact (contactPromptResult) {
+      var isValidated = true;
+
+      if (!contactPromptResult.contact) {
+        contactPromptResult.showErrorMessageFor(
+          'contactSelection',
+          PeoplesTabMessageConstants.CONTACT_NOT_SELECTED_MESSAGE
+        );
+        isValidated = false;
+      } else if (checkContactIsClient(contactPromptResult.contact.id)) {
+        contactPromptResult.showErrorMessageFor(
+          'contactSelection',
+          PeoplesTabMessageConstants.CONTACT_CANT_HAVE_ROLE_MESSAGE
+        );
+
+        isValidated = false;
+      } else if (contactPromptResult.role.role === ts('Client') &&
+        $scope.roles.getActiveNonClientContacts().indexOf(contactPromptResult.contact.id) !== -1) {
+        contactPromptResult.showErrorMessageFor(
+          'contactSelection',
+          PeoplesTabMessageConstants.ROLES_CANT_BE_ASSIGNED_AS_CLIENTS
+        );
+        isValidated = false;
+      }
+
+      if (
```

Comment:
There should be an `else if` here, right? If that is the case, also is tempting to use "return early pattern" here.

Reply:
No @lisandro-compucorp . Because if we put it as `else if`, then only one of the validation error will be shown. But we want both of them. 
![2021-04-28 at 12 58 PM](https://user-images.githubusercontent.com/5058867/116364070-6eb9f580-a821-11eb-84f3-160f023008d7.png)

If I make it elseif, only one of them will be shown.

Reply:
ah, great, I understand, thanks